internal/pkg/admission/policy-server-secret.go

```diff
@@ -22,8 +28,66 @@ func (r *Reconciler) reconcileSecret(ctx context.Context, secret *corev1.Secret)
 	return fmt.Errorf("error reconciling policy-server Secret: %w", err)
 }
 
-type generateCAFunc = func() ([]byte, *admissionregistration.KeyPair, error)
-type pemEncodeCertificateFunc = func(certificate []byte) ([]byte, error)
+func (r *Reconciler) fetchOrInitializePolicyServerSecret(ctx context.Context, policyServerName string, caSecret *corev1.Secret, generateCert generateCertFunc) (*corev1.Secret, error) {
+	policyServerSecret := corev1.Secret{}
+	err := r.Client.Get(
+		ctx,
+		client.ObjectKey{
+			Namespace: r.DeploymentsNamespace,
+			Name:      constants.PolicyServerSecretNamePrefix + policyServerName},
+		&policyServerSecret)
+	if err != nil && apierrors.IsNotFound(err) {
+		return r.buildPolicyServerSecret(policyServerName, caSecret, generateCert)
+	}
+	policyServerSecret.ResourceVersion = ""
+	if err != nil {
+		return &corev1.Secret{},
+			fmt.Errorf("cannot fetch or initialize Policy Server secret: %w", err)
+	}
+
+	return &policyServerSecret, nil
+}
+
+func (r *Reconciler) buildPolicyServerSecret(policyServerName string, caSecret *corev1.Secret, generateCert generateCertFunc) (*corev1.Secret, error) {
+	ca, err := extractCaFromSecret(caSecret)
+	servingCert, servingKey, err := generateCert(
+		ca.CaCert,
+		fmt.Sprintf("%s.%s.svc", constants.PolicyServerServiceNamePrefix+policyServerName, r.DeploymentsNamespace),
```

Comment:
Maybe we can move the `constants.PolicyServerServiceNamePrefix+policyServerName` into a dedicated helper function and just call it from the different parts of the code that are already using this code.

This makes everything more consistent and easier to change in the future, just my 2 cents